scripts/ai/AIDriveStrategyChopperCourse.lua

```diff
@@ -0,0 +1,782 @@
+                            CpUtil.getName(vehicle), unloaderFillLevelPercentage, unloaderDistance, unloaderEte, score)
+                    if score > bestScore then
+                        bestUnloader = vehicle
+                        bestScore = score
+                        bestEte = unloaderEte
+                    end
+                else
+                    self:debug('findUnloader: %s serving my field but already busy', CpUtil.getName(vehicle))
+                end
+            else
+                self:debug('findUnloader: %s is not serving my field', CpUtil.getName(vehicle))
+            end
+        end
+    end
+    if bestUnloader then
+        self:debug('findUnloader: best unloader is %s (score %.1f, ETE %.1f)',
+                CpUtil.getName(bestUnloader), bestScore, bestEte)
+        return bestUnloader, bestEte
+    else
+        self:debugSparse('findUnloader: no idle unloader found')
+    end
+end
+
+-- Copied from parent removed pocket logic and other things that were causes logic issues
+--- Are we ready for an unloader?
+--- @param noUnloadWithPipeInFruit boolean pipe must not be in fruit for unload
+function AIDriveStrategyChopperCourse:isReadyToUnload(noUnloadWithPipeInFruit)
+    -- no unloading when not in a safe state (like turning)
+    -- in these states we are always ready
+    if self:willWaitForUnloadToFinish() then
+        self:debugSparse('isReadyToUnload(): willWait')
+        return true
+    end
+
+    -- but, if we are full and waiting for unload, we have no choice, we must be ready ...
+    if self.state == self.states.UNLOADING_ON_FIELD and self.unloadState == self.states.WAITING_FOR_UNLOAD_ON_FIELD then
+        self:debugSparse('isReadyToUnload(): state')
+        return true
+    end
+
+
+    if not self.course then
+        self:debugSparse('isReadyToUnload(): has no fieldwork course')
+        return false
+    end
+
+    -- around a turn, for example already working on the next row but not done with the turn yet
+
+    if self.course:isCloseToNextTurn(10) then
+        self:debugSparse('isReadyToUnload(): too close to turn')
+        return false
+    end
+    -- safe default, better than block unloading
+    self:debugSparse('isReadyToUnload(): defaulting to ready to unload')
+    return true
+end
+
+function AIDriveStrategyChopperCourse:getConnectingTrack()
+    return self.state == self.states.ON_CONNECTING_TRACK
+end
+
+
+-- Copied from parent altered to remove pipe in fruit map logic. We don't have a pipeInFruit map for choppers
+--- We calculated a waypoint to meet the unloader (either because it asked for it or we think we'll need
+--- to unload. Now make sure that this location is not around a turn or the pipe isn't in the fruit by
+--- trying to move it up or down a bit. If that's not possible, just leave it and see what happens :)
+function AIDriveStrategyChopperCourse:findBestWaypointToUnloadOnUpDownRows(ix, isPipeInFruitAllowed)

```

Comment:
function name does not seem to describe what this is used for here (finding a waypoint to switch unloaders)